### PR TITLE
[produce] 2/2 produce create mac app

### PR DESCRIPTION
**Pull request by hjanuschka**
# PR 2/2

## Those PR's try to add support for creating mac apps with produce.
  * https://github.com/fastlane/fastlane/pull/7180 (adding platform to spaceship)
  * https://github.com/fastlane/fastlane/pull/7181 (adding platform to produce)

## Issues related:
  * https://github.com/fastlane/fastlane/issues/7166  

### run  with:

  * Commandline
```bash
produce -a hjanuschka.macapp2 --platform osx
```

  * Fastfile

```ruby
lane :produce_macapp do
    produce(platform: "osx", app_identifier: "hjanuschka.macapp2")
end
```

there is also a branch with both prs applied: https://github.com/hjanuschka/fastlane/tree/produce_macapp_combined -

### CODE REVIEW

File: produce/lib/produce/options.rb [hunk 1]
```diff
@@ -38,6 +38,15 @@ def self.available_options
...
                                      default_value: Time.now.to_i.to_s,
                                      is_string: true),
+        FastlaneCore::ConfigItem.new(key: :platform,
+                                     short_option: "-j",
+                                     env_name: "PRODUCE_PLATFORM",
+                                     description: "The platform to use (optional)",
+                                     optional: true,
+                                     default_value: "ios",
+                                     verify_block: proc do |value|
+                                                     UI.user_error!("The platform can only be ios or osx") unless %('ios', 'osx').include? value
```
Comment (KrauseFx) [inline produce/lib/produce/options.rb]: Would be great to auto-convert the value to a string, in case the user passes a symbol `platform: :ios` 👍

Comment (KrauseFx) [inline produce/lib/produce/options.rb]: Never mind, as `is_string` is true by default AFAIK

**Comment by milch** [comment]
Just tried it, works well - thanks for your contribution! 🚀

**Review by milch — APPROVED** [review]
🦄

**Pull request merged**